iothub/device/src/HsmAuthentication/HttpClientHelper.cs

```diff
@@ -33,7 +33,8 @@ public static HttpClient GetHttpClient(Uri providerUri)
 #if !NET451
             if (providerUri.Scheme.Equals(UnixScheme, StringComparison.OrdinalIgnoreCase))
             {
-                client = new HttpClient(new HttpUdsMessageHandler(providerUri));
+                using var httpUdsMessageHandler = new HttpUdsMessageHandler(providerUri);
+                client = new HttpClient(httpUdsMessageHandler);
```

Comment:
shouldn't this `HttpUdsMessageHandler` be disposed only when we dispose the `HttpClient`?

Reply:
it will be, as long as it's in scope, it won't be disposed.